adlfs/spec.py

```diff
@@ -1302,6 +1302,20 @@ async def _pipe_file(self, path, value, overwrite=True, **kwargs):
 
     pipe_file = sync_wrapper(_pipe_file)
 
+    async def _cat_file(self, path, start=None, end=None, **kwargs):
+        if end is not None:
+            start = start or 0  # download_blob requires start if length is provided.
+            length = end - start
+        else:
+            length = None
+        container_name, path = self.split_path(path)
+        async with self.service_client.get_blob_client(
+            container=container_name, blob=path
+        ) as bc:
```

Comment:
I guess it would be nice if this can be wrapped into `try/except` for catching `ResourceNotFoundError` though there are some more missing places so maybe we can add some sort of decorator do that automatically on all functions, ([an example](https://github.com/iterative/sshfs/blob/eddccf69abe6e867a451df57dff707412c3c6a4b/sshfs.py#L32-L52]) What do you think?

Reply:
Hmmm I could go either way on that. I'm a bit leery of decorators like this, since their behavior can be hard to understand since they're separated from the implementation of the method. But it's difficult to ensure that we properly handle this everywhere.